hedera-node/hapi-utils/src/main/java/com/hedera/node/app/hapi/utils/ethereum/EthTxData.java

```diff
@@ -322,4 +266,113 @@ EthTxData replaceTo(byte[] to) {
                 r,
                 s);
     }
+
+    /**
+     * Encodes the transaction data into a EthTxData according to legacy RLP format.
+     *
+     * @return the encoded transaction data
+     */
+    private static EthTxData populateLegacyEthTxData(RLPItem rlpItem, byte[] rawTx) {
+        List<RLPItem> rlpList = rlpItem.asRLPList().elements();
+        if (rlpList.size() != 9) {
+            return null;
+        }
+
+        byte[] chainId = null;
+        byte[] v = rlpList.get(6).asBytes();
+        BigInteger vBI = new BigInteger(1, v);
+        byte recId = vBI.testBit(0) ? (byte) 0 : 1;
+        if (vBI.compareTo(BigInteger.valueOf(34)) > 0) {
+            chainId = vBI.subtract(BigInteger.valueOf(35)).shiftRight(1).toByteArray();
```

Comment:
What sort of mysterious incantation to get the `chainId` is this?  I mean, I'm used to seeing things like `0x127`, `0x129`, and so on, so this is very strange.

Reply:
I left the logic like it was before and just detach it into another method. Fair to say, didn't want to introduce changes or refactor the current method due to fear of regression.

Reply:
ok, I totally get that, but still, enlighten me: how does this calculation _work_??

Reply:
Here are some debugging notes, but how and why that dark magic works, I still can't explain to you :mage_man: 
```java
byte[] v = rlpList.get(6).asBytes(); 
// v from transaction hash = [2, 119]

BigInteger vBI = new BigInteger(1, v); 
// vBI = 631 (from bytes array above) as binary => 1001110111

byte recId = vBI.testBit(0) ? (byte) 0 : 1; 
// is the bit of index 0 is equal to 1 => 1001110111 => yes => returns 0

if (vBI.compareTo(BigInteger.valueOf(34)) > 0) { // 631 > 34 => true
    chainId = vBI.subtract(BigInteger.valueOf(35)).shiftRight(1).toByteArray();
    // 631 - 35 = 596 = 1001010100 => shiftRight(1) => 100101010 = 298
}
```

Reply:
I'll study that, TY!

Reply:
@david-bakin-sl [That's](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-155.md#specification) the explanation.